### PR TITLE
Pyscaffold update with manual review of alterations

### PR DESCRIPTION
**Pull request by stijnvanhoey**
This PR does not changes the package itself, but does an update on the tooling and CI for development, following the pyscaffold updates. Main change is the usage of [tox](https://tox.readthedocs.io/en/latest/) to automate the docs, unit tests and dev environment setup, e.g. `tox` instead of `python setup.py test` and `tox -e docs` instead of `python setup.py -e docs`.

I updated the documentation to contribute as well.

### CODE REVIEW

File: .github/CONTRIBUTING.rst [hunk 1]
```diff
@@ -125,10 +125,99 @@ to work on. See also the development guidelines below.
...
+`black <https://black.readthedocs.io/en/stable/index.html>`_. is used to make the code formatting as consistent as possible.The
+required dependencies are included in the development requirements installed when running ````pip install -e .[develop]``.
+
+.. warning::
+   Install and update the ``pre-commit`` hooks before your first git commit to the package!
+
+::
+
+    pre-commit install
+    pre-commit autoupdate
```
Comment (johanvdw) [inline .github/CONTRIBUTING.rst:167]: Looks like this updates the versions to newer versions than .pre-commit-config.yaml. Is this what we want?

File: setup.cfg [hunk 2]
```diff
@@ -60,13 +60,15 @@ exclude =
...
 # Add here test requirements (semicolon/line-separated)
 develop =
+    tox
+    flake8
     pytest
     pytest-cov
     sphinx
     sphinx_rtd_theme
-    recommonmark
     pre-commit
```
Comment (johanvdw) [inline setup.cfg:69]: pre-commit >= 2.9.2 was required or black would not install.

**Review by johanvdw — APPROVED** [review]
Two minor details, not sure if these should be changed.

File: .github/CONTRIBUTING.rst (hunk 1 above)
Comment (stijnvanhoey) [reply .github/CONTRIBUTING.rst:167]: As the autoupdate command actually updates the `pre-commit-config.yaml` file, I do think this is ok to do so. I just make sure the update is done for now (I should autoupdate myself ;-))

Comment (stijnvanhoey) [reply .github/CONTRIBUTING.rst:167]: Updated version in config done

**Pull request merged**